src/trace/implementations/ord_neu.rs

```diff
@@ -863,18 +896,25 @@ mod key_batch {
             if !self.update_stash.is_empty() {
                 // If there is a single element, equal to a just-prior recorded update,
                 // we push nothing and report an unincremented offset to encode this case.
-                if self.update_stash.len() == 1 && self.update_stash.last() == self.result.updates.last() {
+                let time_diff = self.result.times.last().zip(self.result.diffs.last());
+                let last_eq = self.update_stash.last().zip(time_diff).map(|((t1, d1), (t2, d2))| {
+                    let t1 = <<<L as Layout>::TimeContainer as BatchContainer>::ReadItem<'_> as IntoOwned>::borrow_as(t1);
+                    let d1 = <<<L as Layout>::DiffContainer as BatchContainer>::ReadItem<'_> as IntoOwned>::borrow_as(d1);
```

Comment:
Slightly shorter, but I don't care much:
```suggestion
                    let t1 = <<L::TimeContainer as BatchContainer>::ReadItem<'_> as IntoOwned>::borrow_as(t1);
                    let d1 = <<L::DiffContainer as BatchContainer>::ReadItem<'_> as IntoOwned>::borrow_as(d1);
```